src/modules/muorb/slpi/uORBProtobufChannel.cpp

```diff
@@ -44,29 +46,121 @@ static MUORBTestType test_to_run;
 
 fc_func_ptrs muorb_func_ptrs;
 
+// static initialization.
+uORB::ProtobufChannel uORB::ProtobufChannel::_Instance;
+uORBCommunicator::IChannelRxHandler *uORB::ProtobufChannel::_RxHandler;
+std::map<std::string, int> uORB::ProtobufChannel::_AppsSubscriberCache;
+pthread_mutex_t uORB::ProtobufChannel::_rx_mutex = PTHREAD_MUTEX_INITIALIZER;
+pthread_mutex_t uORB::ProtobufChannel::_tx_mutex = PTHREAD_MUTEX_INITIALIZER;
+static bool px4muorb_orb_initialized = false;
+
+int16_t uORB::ProtobufChannel::topic_advertised(const char *messageName)
+{
+	PX4_INFO("Advertising %s on remote side", messageName);
+
+	if (muorb_func_ptrs.advertise_func_ptr) {
+		pthread_mutex_lock(&_tx_mutex);
+		int16_t rc = muorb_func_ptrs.advertise_func_ptr(messageName);
+		pthread_mutex_unlock(&_tx_mutex);
+		return rc;
+	}
+
+	PX4_ERR("advertise_func_ptr is null in %s", __FUNCTION__);
+	return -1;
+}
+
+int16_t uORB::ProtobufChannel::add_subscription(const char *messageName, int32_t msgRateInHz)
+{
+	PX4_INFO("Subscribing to %s on remote side", messageName);
+
+	if (muorb_func_ptrs.subscribe_func_ptr) {
+		pthread_mutex_lock(&_tx_mutex);
+		int16_t rc = muorb_func_ptrs.subscribe_func_ptr(messageName);
+		pthread_mutex_unlock(&_tx_mutex);
+		return rc;
+	}
+
+	PX4_ERR("subscribe_func_ptr is null in %s", __FUNCTION__);
+	return -1;
+}
+
+int16_t uORB::ProtobufChannel::remove_subscription(const char *messageName)
+{
+	PX4_INFO("Unsubscribing from %s on remote side", messageName);
+
+	if (muorb_func_ptrs.unsubscribe_func_ptr) {
+		pthread_mutex_lock(&_tx_mutex);
+		int16_t rc = muorb_func_ptrs.unsubscribe_func_ptr(messageName);
+		pthread_mutex_unlock(&_tx_mutex);
+		return rc;
+	}
+
+	PX4_ERR("unsubscribe_func_ptr is null in %s", __FUNCTION__);
+	return -1;
+}
+
+int16_t uORB::ProtobufChannel::register_handler(uORBCommunicator::IChannelRxHandler *handler)
+{
+	_RxHandler = handler;
+	return 0;
+}
+
+int16_t uORB::ProtobufChannel::send_message(const char *messageName, int32_t length, uint8_t *data)
+{
+	bool is_not_slpi_log = true;
+
+	if ((strcmp(messageName, "slpi_debug") == 0) || (strcmp(messageName, "slpi_error") == 0)
+	    || (strcmp(messageName, "muorb_test") == 0)) { is_not_slpi_log = false; }
```

Comment:
Remove check for muorb_test topic here

Reply:
If we remove this check then send_message will not be called during the 'send_message' test because has_subscribers is set to 0 and is_not_slpi_log is still set to false.

Reply:
There is another comment that says don't call send_message in the test case.